Description: allow increase in c value of ATK_*_LAST_DEFINED
 atk has a number of constants of the form ATK_*_LAST_DEFINED, these can
 increase in new versions of atk. For example ATK_STATE_LAST_DEFINED was
 increased with the upload of atk1.0 2.38 and ATK_ROLE_LAST_DEFINED was
 increased with the upload of at-spi-2-core 2.45 upstream did not bump the
 soversion, so clearly they do not consider this a change that is likely to
 break downstream applictions.

 Futhermore the only rust code I can find that actually uses the
 constants in question is some code in rust-atk that translates them
 to an enum I cannot find any rust code that makes use of the 
 corresponding enum value either.

Author: Peter Michael Green <plugwash@debian.org>

Index: atk-sys/tests/abi.rs
===================================================================
--- atk-sys.orig/tests/abi.rs
+++ atk-sys/tests/abi.rs
@@ -134,6 +134,14 @@ fn cross_validate_constants_with_c() {
             continue;
         }
 
+        if rust_name.starts_with("(gint) ATK_") && rust_name.ends_with("_LAST_DEFINED") {
+            // ATK_*_LAST_DEFINED may increase in newer versions of the atk library.
+            if c_value.parse::<i32>().unwrap_or(0) > rust_value.parse::<i32>().unwrap_or(i32::MAX) {
+                results.record_passed();
+                continue;
+            }
+        }
+
         if rust_value != c_value {
             results.record_failed();
             eprintln!(
